estherh5/api.crystalprism.io
# Migrate data storage to Postgres

- Update .travis.yml file to reflect new environment variables
- Add management.py file to .coveragerc file to omit it from Coverage report
- Add function to load initial database data
- Add backup database and upload to S3 function
- Add schedule weekly backups function
- Update .travis.yml and server.wsgi files to reflect new environment variables
- Add test database to testing utils
- Update .coveragerc file to omit irrelevant files
- Update .travis.yml file to reflect python3.6 testing command
- Update user storage from JSON to database
- Update user tests to reflect database storage change
- Update score storage from JSON to database
- Update score tests to reflect database storage change
- Update drawing and like storage from JSON to database
- Update drawing and drawing like tests to reflect database storage change
- Update post and comment storage from JSON to database
- Update post and comment tests to reflect database storage change
- Update server setup steps for readability
- Add instruction for setting FLASK_APP environment variable
- Add name and size conventions for S3 homepage photos
- Add `/api/homepage/photos` and `/api/homepage/ideas` endpoints [GET] for retrieving homepage photos and thought posts respectively
- Update `/api/thought-writer/posts` endpoint to no longer include webpage owner posts
- Add modified time for posts, comments and users
- Add check for drawing uniqueness
- Add request data validation for user, score, drawing, and post endpoints [POST, PATCH] and management.py functions
- Add email uniqueness check
- Update .gitignore to ignore coverage files
- Update token payload username 
- Update token payload to use database username instead of client-passed username for case sensitivity
- Add post and comment version storage
